fix: modified and added some attributes

bliq00 opened the pull request:
Some attributes were hard-coded.  converted them to variables and added corresponding variable declarations with defaults
Some attributes were missing.  Added them along with corresponding variable declarations with defaults.

bliq00 committed:
fix: modified and added some attributes

Some attributes were hard-coded.  converted them to variables and added corresponding variable declarations with defaults
Some attributes were missing.  Added them along with corresponding variable declarations with defaults.

Apollorion commented:
Thanks for the contribution @bliq00 sorry for the slow response on this - I've added this to my list to review asap. 

Im mostly concerned about adding these to stacks that might not use some of these variables. Like what happens if we send `terraform_external_state_access` to a stack like kubernetes that doesn't have state. So I just want to run this change through a few scenarios before we merge it.

bliq00 commented:
Hi Joey, thanks for reaching out.  No hurry.  I just had a scenario where a
hard-coded value needed to be set to something else and I just took a copy
of the module and modified it for my use. I figured you guys might like
that possible improvement back.  Feel free to make any changes or ask me to
make any changes necessary.  This is my first submission outside of repos I
control so I'm not too familiar with the process yet.

Rob

On Mon, Feb 24, 2025 at 10:09 AM Joey Stout ***@***.***>
wrote:

> Thanks for the contribution @bliq00 <https://github.com/bliq00> sorry for
> the slow response on this - I've added this to my list to review asap.
>
> Im mostly concerned about adding these to stacks that might not use some
> of these variables. Like what happens if we send
> terraform_external_state_access to a stack like kubernetes that doesn't
> have state. So I just want to run this change through a few scenarios
> before we merge it.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/spacelift-solutions/module-stacks/pull/27#issuecomment-2679271656>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACZSKY2L7XQ5T2EWRZTG5VL2RNN35AVCNFSM6AAAAABXBHUUMWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNZZGI3TCNRVGY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
> [image: Apollorion]*Apollorion* left a comment
> (spacelift-solutions/module-stacks#27)
> <https://github.com/spacelift-solutions/module-stacks/pull/27#issuecomment-2679271656>
>
> Thanks for the contribution @bliq00 <https://github.com/bliq00> sorry for
> the slow response on this - I've added this to my list to review asap.
>
> Im mostly concerned about adding these to stacks that might not use some
> of these variables. Like what happens if we send
> terraform_external_state_access to a stack like kubernetes that doesn't
> have state. So I just want to run this change through a few scenarios
> before we merge it.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/spacelift-solutions/module-stacks/pull/27#issuecomment-2679271656>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACZSKY2L7XQ5T2EWRZTG5VL2RNN35AVCNFSM6AAAAABXBHUUMWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNZZGI3TCNRVGY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

jubranNassar commented on main.tf:
```diff
@@ -65,7 +67,8 @@ resource "spacelift_stack" "this" {
 
-  terraform_smart_sanitization = true
+  terraform_external_state_access = var.terraform_external_state_access
+  terraform_smart_sanitization = var.terraform_smart_sanitization
```
```suggestion
  terraform_smart_sanitization = local.is_tf ? var.terraform_smart_sanitization : null
```

jubranNassar commented on main.tf:
```diff
@@ -65,7 +67,8 @@ resource "spacelift_stack" "this" {
   administrative = var.administrative
 
-  terraform_smart_sanitization = true
+  terraform_external_state_access = var.terraform_external_state_access
```
```suggestion
  terraform_external_state_access = local.is_tf ? var.terraform_external_state_access : null
```

jubranNassar commented:
Hey @bliq00  - Thank you for the contribution. I think what I suggested above should resolve the issue for non terraform/tofu stacks.

Apollorion commented:
closing for inactivity, we my re-address these in the future
